WordPress/src/main/java/org/wordpress/android/util/LocaleManager.java

```diff
@@ -0,0 +1,151 @@
+package org.wordpress.android.util;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.res.Configuration;
+import android.content.res.Resources;
+import android.os.Build;
+import android.preference.PreferenceManager;
+
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+/**
+ * Helper class for working with localized strings. Ensures updates to the users
+ * selected language is properly saved and resources appropriately updated for the
+ * android version.
+ */
+public abstract class LocaleManager {
+    private static final String LANGUAGE_KEY = "language-pref";
+
+    /**
+     * Activate the locale associated with the provided context.
+     * @param context The current context.
+     */
+    public static Context setLocale(Context context) {
+        return updateResources(context, getLanguage(context));
+    }
+
+    /**
+     * Change the active locale to the language provided. Save the updated language
+     * settings to sharedPreferences.
+     * @param context The current context
+     * @param language The 2-letter language code (example "en") to switch to
+     */
+    public static Context setNewLocale(Context context, String language) {
+        Locale newLocale = WPPrefUtils.languageLocale(language);
+
+        if (Locale.getDefault().toString().equals(newLocale.toString())) {
+            removePersistedLanguage(context);
+        } else {
+            saveLanguageToPref(context, language);
+        }
+        return updateResources(context, language);
+    }
+
+    /**
+     * Compare the language for the current context with another language.
+     * @param language The language to compare
+     * @return True if the languages are the same, else false
+     */
+    public static boolean isDifferentLanguage(String language) {
+        Locale newLocale = WPPrefUtils.languageLocale(language);
```

Comment:
I know this shouldn't really be a problem (at least not right now), but I am wondering if it's worth to check the emptiness of `language` ourselves instead of letting `WPPrefUtils.languageLocale()` to return the default language to us. Alternatively, it might be even better to just mark the `language` as `@NonNull` so the activity/fragment that's calling this method has to deal with the issue.

To clarify, let's say the current locale is `Turkish` and we somehow passed an empty string to this method, it'll return true since `Turkish` is not the default locale. By handling this case in the method we can return `false` and that'd be good enough. But if we were to change the method to `isSameLocale`, returning `false` in that case is not what we want. So, I feel the `@NonNull` approach is the better one, assuming that it's even worth to do anything here since it will pretty much only prevent developer errors.

Reply:
The default locale is the current locale unless the user is in the process of manually changing the language in which case the language string would not be empty and if it were then we wouldn't want to make a change so we would want it to return the default locale. When a new language is requested a locale object is created from that language and then set as the default locale. When the app is closed and restarts, one of the first few things it does during application startup is to check for the presence of a language override, and if it exists, the default locale is set from it immediately.

Also the `@NonNull` would only check for a null object and not an empty string so it wouldn't gain us anything. 🤔 

Oh, but I did move all the language-related methods out of WPPrefUtils since they were all just helper methods for localization and not really relevant to preferences. It's also nice to keep all those methods in one place so developers can find them in the future.

Reply:
> The default locale is the current locale unless the user is in the process of manually changing the language in which case the language string would not be empty

That sounds good to me, it was just a thought, because some check outside the control of `isDifferentLanguage` was happening in the `WPPrefUtils.languageLocale` which is always a worrisome thing for me. In this specific case, there is no harm and we don't have to do anything here. Thanks for indulging my question.

> Also the @NonNull would only check for a null object and not an empty string so it wouldn't gain us anything.

It doesn't fully fix it, but it covers half the case. Having said that, I was not really sure about if it's good enough either. I think the move to the `LocaleManager` will help with my concern 👍